security/s2a/internal/crypter/counter_test.go

```diff
@@ -0,0 +1,90 @@
+package crypter
+
+import (
+	"math"
+	"testing"
+)
+
+func TestCounterInc(t *testing.T) {
+	for _, tc := range []struct {
+		desc                     string
+		counter, expectedCounter uint64
+		shouldOverflow           bool
+	}{
+		{
+			desc:            "basic 1",
+			counter:         0,
+			expectedCounter: 1,
+		},
+		{
+			desc:            "basic 2",
+			counter:         123,
+			expectedCounter: 124,
+		},
+		{
+			desc:            "almost overflow",
+			counter:         math.MaxUint64 - 1,
+			expectedCounter: math.MaxUint64,
+		},
+		{
+			desc:           "max overflow",
+			counter:        math.MaxUint64,
+			shouldOverflow: true,
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			c := counter{value: tc.counter}
+			c.inc()
+			val, err := c.val()
+			if got, want := err == nil, !tc.shouldOverflow; got != want {
```

Comment:
I know it's not needed but can you change this to `(err == nil)` for clarify? Thanks!